akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -864,6 +864,16 @@ final case class Server(products: immutable.Seq[ProductVersion]) extends jm.head
 // https://tools.ietf.org/html/rfc6797
 object `Strict-Transport-Security` extends ModeledCompanion[`Strict-Transport-Security`] {
   def apply(maxAge: Long, includeSubDomains: Option[Boolean]) = new `Strict-Transport-Security`(maxAge, includeSubDomains.getOrElse(false))
+
+  def fromDirectives(directives: StrictTransportSecurityDirective*) = {
+    val maxAgeDirectives = directives.filter(_.isInstanceOf[MaxAge])
```

Comment:
`directives.collect { case m: MaxAge => m }`, and lift the partial function out to the header object to make sure only a single instance of it is used/needed